Bobsar0/Fast-Food-Fast
# #161858534 Implement email and sms delivery service

#### What does this PR do?
Implement email and sms delivery service

#### Description of Task to be completed?
- Send confirmation email to user after order creation and when order status has been updated by admin.
- Send sms to user upon update in order status

#### How should this be manually tested?
- Clone the repo and run the following commands:
```
  - git checkout ft-email-delivery-#160788274  
  - npm install
  - npm start
```
- Navigate to http://localhost:9999/ and login/create an account 

_You should be redirected to a dedicated menu page_
- Place an order; then check your registered email
_You should see a confirmation email from the app_

To check update status email:
- Login in as an admin with the following credentials:
```
username: admin
password: admin!23
```
- Click on `administrator panel` > `manage orders`
- Search for your order, then `approve`, `decline` or `mark as complete`
- Check your email once again for details with the update order

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/#160788274

#### Background Contexts
This functionality is only enabled during production. To test during development:
- Uncomment relevant lines in ` ./server/api/controllers/ordersDBController` and ` ./server/api/helpers/sms.js`
- Change email address in ./server/api/helpers/email.js to a valid email of your choice
- Create an account with [Twilio](https://twilio.com) and use the free trial number for the sms functionality
- Add a `.env` file with the following details:
```
PASS = _email_password_
SID = _twilio_AcctSID_
AUTH_TOKEN = _twilio_AuthToken_
PHONE = _twilio_free_trial_number_ (with preceding country code)
```
**Note that for some reason, MTN lines don't receive the text messages so check your twilio dashboard console for sms delivery details

#### Screenshots
![image](https://user-images.githubusercontent.com/19636985/48415156-e0368400-e74c-11e8-8bca-d6caf3e99df6.png)
